src_c/polygon.c

```diff
@@ -341,7 +367,90 @@ pg_polygon_copy(pgPolygonObject *self, PyObject *_null)
     return pgPolygon_New(&self->polygon);
 }
 
+static PyObject *
+pg_polygon_move(pgPolygonObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    double x = 0, y = 0;
+    switch (nargs) {
+        case 1:
+            if (PySequence_Check(args[0])) {
+                if (!pg_TwoDoublesFromObj(args[0], &x, &y)) {
+                    return RAISE(PyExc_TypeError,
+                                 "Invalid arguments, must be a sequence of 2 "
+                                 "numbers");
+                }
+            }
+            else if (!pg_DoubleFromObj(args[0], &x)) {
+                return RAISE(PyExc_TypeError, "Polygon.move numeric values");
+            }
+            break;
+        case 2:
+            if (!pg_DoubleFromObj(args[0], &x) ||
+                !pg_DoubleFromObj(args[1], &y)) {
+                return RAISE(PyExc_TypeError,
+                             "Invalid arguments, must be 2 numbers");
+            }
+            break;
+        default:
+            return RAISE(PyExc_TypeError,
+                         "Polygon.move requires 1 or 2 numbers");
+    }
```

Comment:
This whole switch i suppose it's derived from my circle move function, but we got rid of that now. and we do not support passing a single value to move functions, you can just pass them a piar of values. Try to rewrite these like the ones we have now.

Reply:
Done!
Also added an internal function called "move_polygon" to help me out in moving polygon in all of our functions where we need to move. This way will make sure we aren't doing anything wrong as in set center functions.